### DISCUSSION TITLE
Issues while updating to v2.3.1

### DISCUSSION THREAD

**Discussion by srajmane** [bazelbuild/rules_nodejs#2354] [Q&A]
I am updating rules_nodejs version from 0.36.0 to latest 2.3.1.
I have gone through all change log and https://github.com/bazelbuild/rules_nodejs/wiki#migrating-off-build_bazel_rules_nodejsdefsbzl,
Currently, I have made below changes
- Replaced @build_bazel_rules_nodejs//:defs.bzl with @build_bazel_rules_nodejs//:index.bzl
- Replaced @bazel/bazel with @bazel/bazelisk
- Replaced  load("@build_bazel_rules_nodejs//:defs.bzl", "rollup_bundle") --> load("@npm//@bazel/rollup:index.bzl", "rollup_bundle")
- Replaced load("@npm_bazel_jasmine//:index.bzl", "jasmine_node_test") --> load("@npm//@bazel/jasmine:index.bzl", "jasmine_node_test")
- Replaced load("@npm_bazel_typescript//:index.bzl", "ts_library") --> load("@npm//@bazel/typescript:index.bzl", "ts_library")
- Removed `enable_code_splitting = False` 
- Updated package.json file

Still, I am facing issue with,
`missing input file bundle.es2015.js` and `[!] Error: The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten`

how can I fix this error?

Thank you.

**Comment by alexeagle** [comment]
trying out this new GitHub "discussions" feature since you filed an issue but there's probably nothing wrong with rules_nodejs.

Historically we've only done user support on Slack but this seems worth trying ;)

I would suggest you bisect it. You're upgrading over more than a year of changes. Try getting to 1.0 first. Then you'll figure out exactly which version upgrade is causing you problems, and can look through the changelog in more detail for that revision.

**Reply by srajmane** [reply]
@alexeagle I have tried with rules_nodejs 1.0.0 and 1.7.0
But didn't find any changelog related es2015. 
Facing `missing input file bundle.es2015.js` issue. 
Checking with `0.42.3` version.

**Reply by srajmane** [reply]
@alexeagle I tried with 0.40.0, facing below-mentioned issue:
The legacy rollup_bundle rule has been deprecated

As announced in https://github.com/bazelbuild/rules_nodejs/releases/tag/0.38.0,
there is a new rollup_bundle Bazel rule.
Migration instructions are at
https://github.com/bazelbuild/rules_nodejs/wiki#migrating-off-internalrollup

It is failing for bundle.es2015.js rollup_bundle rule. Any pointer for this error?
